src/plugins/opensearch_dashboards_react/public/code_editor/code_editor.tsx

```diff
@@ -127,33 +135,6 @@ export class CodeEditor extends React.Component<Props, {}> {
       this.props.editorWillMount();
     }
 
-    monaco.languages.onLanguage(this.props.languageId, () => {
```

Comment:
this looks like existing code that technically anyone who is using the code_editor component will be getting rid of correct? without explicit opt in then these providers are gunna be set by default which could be a breaking change in some places.

Reply:
its handled in the render() function. The reason I moved this there is because this method was within `_editorWillMount`, so all of this registration would only happen once. This resulted in the editor only registering the first language it sees, and if the language changes, none of these providers will be set for the new language. Functionality is exactly the same upon start.

Reply:
also another way of saying this is that this line:
```
monaco.languages.onLanguage(this.props.languageId, () => {
```
does nothing here

Reply:
have we verify no performance issues? that means anytime this is re-rendered its going to re register

Reply:
i checked this when i made this change, it only triggered on every language switch

Reply:
We'd better move the code to like `componentDidUpdate` and detect if there are changes on languageId.